src/dmd/access.d

```diff
@@ -612,8 +612,24 @@ public Dsymbol mostVisibleOverload(Dsymbol s)
         else
             break;
 
-        if (next && mostVisible.prot().isMoreRestrictiveThan(next.prot()))
+        if (next && mostVisible.effectiveProtAt(mod).isMoreRestrictiveThan(next.effectiveProtAt(mod)))
             mostVisible = next;
     }
     return mostVisible;
 }
+
+/**
+ * Return the "effective" protection attribute of a symbol seen from a module.
+ * The effective protection attribute is the same as the regular protection attribute,
+ * except package() is "private" if the module is outside the package;
+ * otherwise, "public".
+ */
+Prot effectiveProtAt(Dsymbol d, Module mod)
```

Comment:
Please spell this out as "effectiveProtection".  I had to read the comment to understand what "Prot" meant.

Reply:
"protection from" sounds like a D&D spell. :)